### DISCUSSION TITLE
HXH†  = Z; HYH† = −Y ; HZH† = X

### DISCUSSION THREAD

**Discussion by Fe-r-oz** [QuantumSavory/QuantumClifford.jl#285] [Q&A] [answered]
I have a very trivial question:

I want to do this operation: HYH† = -Y. For some quantum codes, we need to apply the adjoint operation or HUH† on the logical operator. In fact, Gottesman uses this trick HXH†  = Z; HYH† = −Y ; HZH† = X  to create a new set of quantum codes by applying the operation H"logicaloperator"H† for each operator in each generator of the stabilizer. 

At the moment, we can do this by manually swapping X -> Z and Z- > X, like checking (true, false), if yes, then (false, true). etc. 

Suppose we want to change the sign of second logical operator which is  Y in S"XYZ" to -Y which is the result when we apply HYH† = -Y.  How can this be done if we don't follow the way of adjoint?

 Is there a way to perform the aforementioned operation either via the current approach or via adjoint (if that is feasible) ?
 
Currently, we can only apply H, not H† 

Currently,

```
Suppose s= S"Y"
julia> s=S"X";

julia> sHadamard(1)*s
+ Z 

julia> s=S"XYZ";

julia> sHadamard(1)*s
+ ZYZ
```

I checked the` sumtypes.jl,` and tried to use adjoint, 

```
Closest candidates are:
  adjoint(::Graphs.DefaultDistance)
   @ Graphs ~/.julia/packages/Graphs/S7tMY/src/distance.jl:24
  adjoint(::Missing)
   @ Base missing.jl:101
  adjoint(::Type{<:QuantumClifford.CompactifiedGate})
   @ QuantumClifford ~/.julia/packages/SumTypes/2baVy/src/sum_type.jl:274
```

Thanks a lot.

**Comment by Krastanov** [comment, marked as answer]
Hi, Feroz!

Pardon the delay in answering, busy week. I am not completely sure I understand the question, so feel free to redirect my explanation.

A few things:

- QuantumClifford does not provide any particularly useful `adjoint` methods for gates
- H and H† are the same gate (true about H, X, Y, Z, CNOT, CPHASE, SWAP, any of the conditional □C□ gates, but not true in general)
- U† is the same as U⁻¹ for any unitary gate
- `LinearAlgebra.inv` methods are defined for `CliffordOperator`, but not for `AbstractSymbolicOperator` (created an issue about that https://github.com/QuantumSavory/QuantumClifford.jl/issues/286) 

```
julia> inv(tCNOT)
X₁ ⟼ + XX
X₂ ⟼ + _X
Z₁ ⟼ + Z_
Z₂ ⟼ + ZZ

julia> inv(sCNOT(1,2))
ERROR: MethodError: no method matching inv(::sCNOT)

julia> inv(CliffordOperator(sCNOT))
X₁ ⟼ + XX
X₂ ⟼ + _X
Z₁ ⟼ + Z_
Z₂ ⟼ + ZZ
```

**Reply by Fe-r-oz** [reply]
No problem at all.  Thanks!
